guard/src/commands/completions.rs

```diff
@@ -0,0 +1,76 @@
+use crate::command::Command;
+use crate::commands::{APP_NAME, APP_VERSION, COMPLETIONS};
+use crate::utils::reader::Reader;
+use crate::utils::writer::Writer;
+use crate::{commands, rules};
+use clap::{Arg, ArgAction, ArgMatches};
+
+#[derive(Copy, Clone, Debug)]
+pub enum Shell {
+    Bash,
+    Zsh,
+    Fish,
+}
+
+impl From<String> for Shell {
+    fn from(value: String) -> Self {
+        match value.as_str() {
+            "bash" => Shell::Bash,
+            "zsh" => Shell::Zsh,
+            "fish" => Shell::Fish,
+            _ => unimplemented!(),
+        }
+    }
+}
+
+#[derive(Default, Debug)]
+pub struct Completions {}
+
+const SHELL_TYPES: [&str; 3] = ["bash", "zsh", "fish"];
+const SHELL: (&str, char) = ("shell", 's');
+
+impl Command for Completions {
+    fn name(&self) -> &'static str {
+        COMPLETIONS
+    }
+
+    fn command(&self) -> clap::Command {
+        clap::Command::new(COMPLETIONS).arg(
+            Arg::new(SHELL.0)
+                .long(SHELL.0)
+                .short(SHELL.1)
+                .required(true)
+                .value_parser(SHELL_TYPES)
+                .action(ArgAction::Set),
+        )
+    }
+
+    fn execute(&self, args: &ArgMatches, _: &mut Writer, _: &mut Reader) -> rules::Result<i32> {
+        let mut app = clap::Command::new(APP_NAME).version(APP_VERSION);
+
+        let commands: Vec<Box<dyn Command>> = vec![
```

Comment:
Will we need to keep this up to date with `main.rs` each time we change a command or add a new one?

Should we create a method and reuse it instead?